source/agora/consensus/state/ValidatorSet.d

```diff
@@ -783,6 +783,41 @@ public class ValidatorSet
 
         return ex_validators;
     }
+
+    /***************************************************************************
+
+        Query stakes of active Validators
+
+        Params:
+            peekUTXO =
+            utxos =
+
+        Returns:
+            
+
+    ***************************************************************************/
+
+    public UTXO[] getValidatorStakes (UTXOFinder peekUTXO, out UTXO[] utxos)
```

Comment:
ditto comment about `ref` and `assumeSafeAppend`. And I suggest changing it in the EnrollmentManager too.